### PR TITLE
Polish test case for PR: replace `into-stream` to `Readable.from`

### PR DESCRIPTION
**Pull request by stanleyxu2005**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by stanleyxu2005** [comment]
I'd like to polish the test case @climba03003 and @mcollina create. But it's strage that your previous PR and this PR are failing at my local env.

**Comment by stanleyxu2005** [comment]
If I replace `unidici` with `axois`, it works
```js
  // const response1 = await request(`${address}/compress`)
  // const response2 = await request(`${address}/no-compress`)
  // const body1 = await response1.body.text()
  // const body2 = await response2.body.text()
  const body1 = (await axios(`${address}/compress`)).data
  const body2 = (await axios(`${address}/no-compress`)).data
  t.equal(body1, body2)
  t.equal(body1.length, twoByteUnicodeContent.length)
  t.equal(body2.length, twoByteUnicodeContent.length)
```

**Comment by gurgunday** [comment]
That's interesting

node 14 doesn't support undici fetch I think... you might want to try `node-fetch@2` if you'd like to use it here

**Comment by climba03003** [comment]
I don't understand why it is strange on your environment. The test case works fine between different OS and Node versions. 

I have also tested on my local environment.
What's the problem of `undici.request` ?

**Comment by stanleyxu2005** [comment]
> I don't understand why it is strange on your environment. The test case works fine between different OS and Node versions.
> 
> I have also tested on my local environment. What's the problem of `undici.request` ?

Before I removed `unidici`, received response text lengths are not identical. You can check the attached screenshot and also the first couple of pipeline failure reports.

**Comment by gurgunday** [comment]
> I don't understand why it is strange on your environment. The test case works fine between different OS and Node versions.
> 
> I have also tested on my local environment. What's the problem of `undici.request` ?

@climba03003 I was going to ask if you had an idea on why `undici.request` might be returning a different response text length, maybe there is some post processing based on the charset of the response?

So request didn't work: https://github.com/fastify/fastify-compress/pull/291/commits/ffe60f61defc6a70e5015e9b9ecfa355d606b6b4

And fetch did: https://github.com/fastify/fastify-compress/pull/291/commits/c1882b91b6044f45c197cf8562ece0133ea5919d

File: package.json [hunk 1]
```diff
@@ -23,11 +23,11 @@
     "adm-zip": "^0.5.10",
     "fastify": "^4.19.2",
     "jsonstream": "^1.0.3",
+    "node-fetch": "2",
```
Comment (gurgunday) [inline package.json:26]:
```suggestion
    "node-fetch": "^2",
```

**Comment by stanleyxu2005** [comment]
> I don't understand why it is strange on your environment. The test case works fine between different OS and Node versions.
> 
> I have also tested on my local environment. What's the problem of `undici.request` ?

Here are my thoughts of failing or not failing @climba03003 

(1) In my attached img, the input length (334k) and output length (284k) are not identical.

(1) The previous PR you created, only checks the responses of compressed and uncompressed. They are identical on supported platforms (including my dev env1 - Win11), but for some reasons, not identical on my dev env2 - a legacy Windows.

(2) In my PR, I check the input and output length in additional. That's why it is failed. 

(3) Why the input and output lengths are not same, maybe there are some hidden settings for encoding. I'm not familiar with `unidici`, so I cannot tell, if it is a bug of `unidici` or not.

image

**Review by gurgunday — APPROVED** [review]
lgtm

**Review by mcollina — CHANGES_REQUESTED** [review]
Please use use my test. undici.fetch() correctly decode this. undici.request() for some reason does not. I'll open an issue.

Use undici@5.

**Comment by gurgunday** [comment]
@mcollina undici fetch doesn’t work in node 14 I think, do you know if that’s the case?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
You are right, there is no web streams in node v14.

**Pull request merged**